public/src/pages/auth/AdminPage.tsx

```diff
@@ -1,23 +1,52 @@
 import React, { FC, useEffect, useState } from 'react';
-import { getAdminStatus } from '../../api/httpClient';
-import AuthLayout from './AuthLayout';
+import { UserData } from 'src/interfaces/User';
+import { getAdminStatus, searchUsers } from '../../api/httpClient';
+import AdminLayout from './AdminLayout';
 
 export const AdminPage: FC = () => {
   const user = sessionStorage.getItem('email') || localStorage.getItem('email');
   const [isAdmin, setIsAdmin] = useState<boolean>(false);
+  const [inputValue, setInputValue] = useState('');
+  const [users, setUsers] = useState<UserData[]>([]);
+
+  const onInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
 
   useEffect(() => {
-    if (user) getAdminStatus(user).then((data) => setIsAdmin(!!data.isAdmin));
-  }, [user]);
+    if (user) {
+      getAdminStatus(user).then((data) => setIsAdmin(!!data.isAdmin));
+      inputValue.length
+        ? searchUsers(inputValue).then((data) => setUsers(data))
+        : setUsers([]);
```

Comment:
I'm not sure what you mean here. IMO if the input is empty, you have to display all users

Reply:
Hm, ok, agreed with you